buildpack/build_test.go

```diff
@@ -349,19 +349,37 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 			})
 
 			when("the launch, cache and build flags are false", func() {
-				it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
-					h.Mkdir(t,
-						filepath.Join(layersDir, "A", "layer"),
-					)
-					h.Mkfile(t,
-						"[types]\n  build=false\n  cache=false\n  launch=false",
-						filepath.Join(layersDir, "A", "layer.toml"),
-					)
+				when("the flags are specified in <layer>.toml", func() {
+					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
+						h.Mkdir(t,
+							filepath.Join(layersDir, "A", "layer"),
+						)
+						h.Mkfile(t,
+							"[types]\n  build=false\n  cache=false\n  launch=false",
+							filepath.Join(layersDir, "A", "layer.toml"),
+						)
 
-					_, err := bpTOML.Build(buildpack.Plan{}, config)
-					h.AssertNil(t, err)
-					h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
-					h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
+						_, err := bpTOML.Build(buildpack.Plan{}, config)
+						h.AssertNil(t, err)
+						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
+						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
+					})
+				})
+
+				when("the flags aren't specified in <layer>.toml", func() {
+					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
+						h.Mkdir(t,
+							filepath.Join(layersDir, "A", "layer"),
+						)
+						h.Mkfile(t,
+							filepath.Join(layersDir, "A", "layer.toml"),
```

Comment:
I think calling `h.Mkfile()` with one argument is a no-op, as it ranges over `paths`. That is interesting to know that the test is still passing but I think it would be more accurate to create the file with blank data.

Reply:
Thanks! Please see https://github.com/buildpacks/lifecycle/pull/554/commits/6d5219299b503d794427d6ae41f23a35ac0c6d99.